website/src/services/lapisClient.ts

```diff
@@ -53,6 +53,16 @@ export class LapisClient extends ZodiosWrapperClient<typeof lapisApi> {
         });
     }
 
+    public getSequenceEntryVersionDetailsTsv(accessionVersion: string) {
+        return this.call('details', {
+            [this.schema.primaryKey]: accessionVersion,
+            dataFormat: 'TSV',
+            // This type cast isn't pretty, but if the API would be typed correctly, the union type
+            // of the actual details resonse and the potential 'string' would polute the whole API,
+            // so I decided to just do this cast here. We know that the return value is a TSV string.
+        }).then((result) => result.map((data) => data as unknown as string));
```

Comment:
I would appreciate input on this. I have fiddled with the Zodios setup quite a bit, but reached the conclusion that this would be the most sensible way to handle the TSV data. Alternatives I considered:

- Get the JSON data from the API and turn it into TSV in somewhere in the backend server (not necessarily in this function). Pros: no casting required. Cons: duplicating the TSV functionality.
- Tried to modify the Zodios API definition and adjusted the Type so it can return a string as well. Pro: Typesafety possible. Cons: the 'string' edgecase polutes all the other regular usages of the endpoint and makes the code much more convoluted in many places.

I think with Zodios the _best_ solution would be to distinguish json and TSV via endpoint, i.e. use a different endpoint alltogether like (`sample/details/tsv` - I know, not very RESTy but you get the idea). This way we could have a nicely typed TSV endpoint. But that requires a backend API change.

Reply:
I'm not the right person to comment on this - I don't know my zodios. @fengelniederhammer might have ideas